### PR TITLE
feat(db): Add GORM with MySQL

### LINKED ISSUE #2: Integrate MySQL database.
* Add MySQL database to the server using gorm.

### PR DESCRIPTION
**Pull request by SuhailAhmed2627**
- Added MySQL Database with Gorm
- Configured config file
- Added Validator for routes

### CODE REVIEW

**Comment by SuhailAhmed2627** [comment]
Closes #2

File: config/init.go [hunk 1]
```diff
@@ -64,8 +44,4 @@ func InitConfig() {
 	default:
 		fmt.Println(color.RedString("Error setting current config"))
 	}
-}
-
-func GetConfig() Config {
-	return currentConfig
-}
+}
```
Comment (Muhesh7) [inline config/init.go]: EOF
Add a new line at the end of the file

File: controllers/user/get_user.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+		return utils.SendResponse(c, http.StatusBadRequest, "User not found")
+	}
+
+	res := GetUserResponse{
+		Username: user.Username,
+		Name:     user.Name,
+	}
+
+	return utils.SendResponse(c, http.StatusOK, res)
+}
```
Comment (Muhesh7) [inline controllers/user/get_user.go]: EOF

File: database/connect.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+)
+
+// ConnectDB connect to db
+func ConnectDB() {
+
+	config := config.GetConfig()
+
+	var err error
+
+	dsn := config.DBUsername + ":" + config.DBPassword + "@tcp(" + config.DBHost + ":" + config.DBPort + ")/" + config.DBName + "?charset=utf8mb4&parseTime=True&loc=Local"
```
Comment (Muhesh7) [inline database/connect.go]: https://freshman.tech/snippets/go/string-concatenation/
above doc has some other efficient ways of concatenating strings in go
kindly refer to it.

**Review by Muhesh7 — CHANGES_REQUESTED** [review]

**Review by Muhesh7 — APPROVED** [review]
Enhancement PR merged
hacktoberfest-accepted

**Pull request merged**